Live metrics docs (#4012)
iterative/dvc.org: content/docs/studio/how-it-works.md

# How Iterative Studio Works

> [Iterative Studio](https://studio.iterative.ai/) works very closely with your
> Git ecosystem. It parses the data, metrics, hyperparameters and model metadata
> that you add to your ML project Git repositories, and renders them on a web UI.

Reviewer: This change is unrelated in this PR, but since it was updated, here's a suggestion because the current wording is unclear, esp. around the concept of metadata:

```suggestion
[Iterative Studio](https://studio.iterative.ai/) works very closely with your
Git ecosystem. It parses the metadata found in your ML project repositories to
present organized data, hyperparameters, models, and metrics on a web UI.
```

> It also parses and displays live metrics and plots sent by [DVCLive]; these
> don't need to be pushed to the Git repositories.

Reviewer: I think this is included in the "and metrics" part. We didn't even mention DVC earlier in this paragraph so why mention DVCLive? Too specific. The wording I suggested above also prevents the need to clarify about not having to push to Git. So I'd remove it from here
```suggestion
```
But maybe we can add this note somewhere else further down... Let me 👀

Author: We need to mention this here coz otherwise it's not clear that live metrics are not "found in your ML project repositories".

> The following video illustrate how Iterative Studio works with data from your ML
> projects and experiments.
> 
> > Note that we have renamed DVC Studio to Iterative Studio and Views to
> > Projects.
> 
> https://www.youtube.com/watch?v=5xM5az78Lrg

Reviewer: The video and it's purpose (being in this place of the doc) should be self-explanatory. Also, proper admonition:

```suggestion
https://www.youtube.com/watch?v=5xM5az78Lrg

<admon>

We have renamed DVC Studio to Iterative Studio and Views to Projects.

</admon>
```

Author: Am not making this change, coz we use this same format in 3 places where we have videos; this note applies to the videos specifically.

## How you save your project data

> - Using [DVC] and [Git], you will push all your ML experiments to your GitHub,
>   GitLab or Bitbucket repositories as Git commits.

Reviewer: 💅🏼 
```suggestion
- Using [DVC] and [Git], you will push all your ML experiments to your GitHub,
  GitLab or Bitbucket repositories.
```

- Using Iterative Studio, or using the command line interface (CLI) of [GTO],
  and possibly [MLEM], you will push all your ML model details to the Git

>   repositories as Git commits and Git tags.

Author: We don't support other artifacts in Studio. And it is important to mention that the info is saved in Git tags. I'll reword this.

>   repositories as Git commits and Git tags.

Author: I've reworded it.

> - Using [DVCLive], you will push live metrics and plots from your Git-committed
>   experiments to Iterative Studio.

Reviewer: 💅🏼 minor (just a bit shorter)
```suggestion
- Using [DVCLive], you will push live metrics and plots directly to Iterative
  Studio.
```

Author: I think it is important to mention that the experiments are committed to Git, coz we don't support live metrics for local experiments yet.

Author: > from your experiments running on a remote machine.

This is still confusing. We need the experiments to have been submitted from Studio and committed to Git. `Remote machine` could be any remote machine - for me, if I do some work in a cloud instance, then I'm working on a remote machine coz it's not my local machine.

I've updated the text, but differently as shown below:

`Using [DVCLive], you will push live metrics and plots from your experiments to Iterative Studio. Note that at the moment, live metrics are supported only for experiments that you [submit from the Iterative Studio UI].`

Let me know if it still doesn't make sense.

Reviewer 2: It makes more sense now, thanks!

## How Iterative Studio extracts your project data

- When you connect to the Git repositories from Iterative Studio, the project's
  `dvc.yaml` is used to identify all the data, metrics and hyperparameters in
  your experiments.
- If you are not using DVC, you can
  [add the metrics and hyperparameters to your Git repositories manually](/doc/studio/user-guide/projects-and-experiments/configure-a-project#custom-metrics-and-parameters).
- Details of your ML models, including versions and stage assignments, are
  extracted from the Git commits and tags.
- Live metrics and plots for experiments committed to Git are extracted from the

>   data you send using [DVCLive].

Reviewer: Wait. I thought live metrics/plots were not committed/pushed to Git? The mechanics start getting confusing here.

Author: The live metrics/plots aren't committed to Git. But the experiments are (there has to be a Git commit that triggers the experiment).

Author: IMO, `remote` is not enough, coz it does not imply that there is a Git commit for it. Users could interpret `remote` to mean any (cloud) machine that is not your laptop/desktop.

Reviewer: I see 2 issues here:

- Emphasize where experiments run/are saved (local vs remote) and what that means to the user from the PoV of the Studio UI
- Differentiate (or not) between experiments and commits (long-standing DVC discussion). I incline to call the first "experiments" and the latter "commits" or "project versions" (even if we clarify that they used to be exps.

Author: @jorgeorpinel We've now already merged the docs for live metrics for local experiments. Can you pls read that version and comment if you still see issues?

Reviewer: The only part that confuses me a little now is under https://dvc.org/doc/studio/how-it-works#how-iterative-studio-runs-experiments-and-model-actions:

> ### How Iterative Studio runs experiments...
>
> - When you [run new experiments](https://dvc.org/doc/studio/user-guide/projects-and-experiments/run-experiments) ... in Iterative Studio, it creates Git commits

So are experiments commits? (Could be what a reader wonders.)

Author: Yes, the experiments are pushed to Git, as Git commits and possibly PRs (if the change was submitted to a new branch).

# How Iterative Studio Works

[Iterative Studio](https://studio.iterative.ai/) works very closely with your
Git ecosystem. It parses the data, metrics, hyperparameters and model metadata
that you add to your ML project Git repositories, and renders them on a web UI.

Additionally, you can [track metrics from your experiments in real
time][live-metrics-and-plots] by pushing the metrics directly to Iterative

> Studio using [DVCLive].

Reviewer: This seems like a general description. Why does it now link specifically to the one specific feature but nothing else from the list in the first paragraph? I think we could keep it general here e.g. avoid the new paragraph, just add "...parses the data and metrics in real time, hyperparams..."

Reviewer: Thanks

The following two videos illustrate how Iterative Studio works with data from

> your ML projects and experiments.

Reviewer: Minor but I'm also not sure we want 2 video embeds one after another here.

We have something similar here: https://dvc.org/doc/user-guide/experiment-management/experiments-overview#work-with-dvc-experiments-from-a-gui and it's a bit strange.

- You can set up your CI/CD actions (e.g. GitHub Actions) to run model training
  upon the creation of Git commits, tags or pull requests. You can use [CML] in
  your CI/CD actions for continuous machine learning.
- You can use [DVCLive] in your CI/CD actions to [send live metrics of model

>   training to Iterative Studio][live-metrics-and-plots].

Reviewer: Hmmm, how are these things that Studio does? (This is under **How Iterative Studio runs ... model actions**.)

Author: I've merged this point with the previous one; it should be better now.

  changes.
- You can set up your CI/CD actions (e.g. GitHub Actions) to run model training
  upon the creation of Git commits, tags or pull requests. You can use [CML] in
  your CI/CD actions for continuous machine learning. And you can use [DVCLive]

>   to [send live metrics of model training to Iterative

Author: Done.
